packages/shared/dao/index.ts

```diff
@@ -156,70 +155,72 @@ export function commsStatusUrl(domain: string, includeLayers: boolean = false, i
   const queryParameters: string[] = []
 
   if (includeLayers) {
-    queryParameters.push('includeLayers=true')
+    queryParameters.push("includeLayers=true")
   }
 
   if (includeUsersParcels) {
-    queryParameters.push('includeUsersParcels=true')
+    queryParameters.push("includeUsersParcels=true")
   }
 
   if (queryParameters.length > 0) {
-    url += '?' + queryParameters.join('&')
+    url += "?" + queryParameters.join("&")
   }
 
   return url
 }
 
 export async function fetchCatalystStatuses(nodes: { domain: string }[]) {
-  const results: PingResult[] = await Promise.all(nodes.map((node) => ping(commsStatusUrl(node.domain, true, true))))
-
-  const realmsWithCapacity = results.filter((realm) => (realm.result?.maxUsers ?? 0) > (realm.result?.usersCount ?? -1))
-
-  return zip(nodes, realmsWithCapacity).reduce(
-    (union: Candidate[], [{ domain }, { elapsed, result, status }]: [CatalystNode, PingResult]) => {
-      function buildBaseCandidate() {
-        return {
-          catalystName: result!.name,
-          domain,
-          status: status!,
-          elapsed: elapsed!,
-          lighthouseVersion: result!.version,
-          catalystVersion: result!.env.catalystVersion
-        }
+  const results = (
+    await Promise.all(
+      nodes.map(async (node) => ({ domain: node.domain, ...(await ping(commsStatusUrl(node.domain, true, true))) }))
+    )
+  ).filter((realm) => (realm.result?.maxUsers ?? 0) > (realm.result?.usersCount ?? -1))
```

Comment:
we could add this filter inside the `.reduce` fn, like we do when `status !== ServerConnectionStatus.OK`

if (condition) { return union }

Reply:
I'd say having the filter before the reduce is much simpler to read, but i can add it in the reduce